src/pages/KubernetesError.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div>
+    <div class="page-body">
+      <h2 data-test="k8s-error-header">
+        <img id="logo" src="../../resources/icons/logo-square-red@2x.png" />
+        Kubernetes Error
+      </h2>
+      <div class="k8s-error">
+        <div class="error-part">
+          <h4>{{ titlePart }}</h4>
+          <p>{{ mainMessage }}</p>
+        </div>
+        <div class="error-part">
+          <h4>Last command run:</h4>
+          <p>{{ lastCommand }}</p>
+        </div>
+        <div v-if="lastCommandComment" class="error-part">
+          <h4>Context:</h4>
+          <p>{{ lastCommandComment }}</p>
+        </div>
+        <div v-if="logLines.length" class="error-part">
+          <h4>Some recent logfile lines:</h4>
+          <pre id="log-lines">{{ wrappedLines }}</pre>
+        </div>
+      </div>
+    </div>
+    <footer class="page-footer">
+      <div class="button-area">
+        <button data-test="accept-btn" class="role-primary" @click="close">
+          Close
+        </button>
+      </div>
+    </footer>
+  </div>
+</template>
+
+<script lang="ts">
+
+import { ipcRenderer } from 'electron';
+import Vue from 'vue';
+import wrap from 'word-wrap';
+
+export default Vue.extend({
+  layout: 'dialog',
+  data() {
+    return {
+      titlePart:          '',
+      mainMessage:        '',
+      lastCommand:        '',
+      lastCommandComment: '',
+      logLines:           [],
+    };
+  },
+  computed: {
+    wrappedLines(): string {
+      const leadingWSPtn = /^(\s+)(.+)$/;
+      const indent = '    ';
+
+      return this.logLines.map((line) => {
+        // word-wrap is a bit brain-dead: either you get no leading indent, or you get it on all lines
+        const m = leadingWSPtn.exec(line);
+        const [leadingWS, rest] = m ? [m[1], m[2]] : ['', line];
+        const fixedLine = wrap(rest, {
+          width: 60,
+          indent
+        });
+
+        return leadingWS + fixedLine.trimStart();
+      }).join('\n');
+    }
+  },
+  mounted() {
+    ipcRenderer.send('kubernetes-errors/ready');
+    ipcRenderer.on('kubernetes-errors-details', (event, titlePart, mainMessage, lastCommand, lastCommandComment, logLines) => {
```

Comment:
Not sure JS works that way -- I thought it might be handling the same message until it gets to the end, where the message-handler for `...-details` is defined. Then the handler for `.../ready` will fire, and if it sends a `...-details` event we'll be ready for it. But, sure, it doesn't hurt to reverse the order.